### PR TITLE
stm32: add support for the STM32F4Discovery

### PR DESCRIPTION
**Pull request by mickey-happygolucky**
Support STM32F4Discovery

This is experimental support.

[Peripherals Support]
* LED(GPIO)
* UART(USART2)

[settings limitations]
* SYSCLK is fixed to 168mhz
* USART2 configurations are fixed to 115200 8N1

I tested examples as follows:

* blinky1
* blinky2
* serial
* echo
* test

### CODE REVIEW

**Comment by deadprogram** [comment]
Hi @mickey-happygolucky can you please add a "smoke" test which is currently just making sure that the board can be built for. Example https://github.com/tinygo-org/tinygo/blob/dev/.circleci/config.yml#L70

File: src/machine/board_stm32f4disco.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	PH1  = portH + 1
+)
+
+const (
+	LED  = LED_BUILTIN
+	LED1 = LED_GREEN
+	LED2 = LED_ORANGE
+	LED3 = LED_RED
+	LED4 = LED_BLUE
+	LED_BUILTIN = LED_GREEN
```
Comment (aykevl) [inline src/machine/board_stm32f4disco.go]: Looks like you didn't gofmt the code. Can you run `go fmt` over it? The command `make fmt` runs it over the entire project.
(There is #275 to improve this in general: some other code was accidentally left unformatted).

File: src/machine/board_stm32f4disco.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
+// +build stm32,stm32f4disco
+
+package machine
+
+// https://wiki.stm32duino.com/index.php?title=File:Bluepillpinout.gif
```
[the remark this answers is not in the document]
Comment (mickey-happygolucky) [reply src/machine/board_stm32f4disco.go]: I tried to delete `package machine` line.
Then I got error as follows,

```
error: /home/mickey/go/src/github.com/tinygo-org/tinygo/src/machine/board_stm32f4disco.go:4:1: expected 'package', found 'const' (and 1 more errors)
```

I think it is necessary, Can you give me an opinion?

[the remark this answers is not in the document]
Comment (aykevl) [reply src/machine/board_stm32f4disco.go]: I meant the comment, which refers to the blue pill board. The `package machine` is required, this is a normal Go package.

[the remark this answers is not in the document]
Comment (mickey-happygolucky) [reply src/machine/board_stm32f4disco.go]: Thanks!
I understood. I will fix.

**Comment by aykevl** [comment]
Thank you a lot for this work! I've merged your changes in f5e5169d6bd4769b8d5b9f1baebb11392133314d.

**Pull request closed without merging**